Gc login page

gchodancar19 opened the pull request:
Created the login page.

gchodancar19 committed:
login page started

gchodancar19 committed:
made more pages and done styling

Jforjo reviewed (changes requested):
This code will not run as their are multiple variables that do not exist: $db_connect for example. We are also not using MYSQLI commands to connect to the database and run SQL commands. We worked on this in class and you should commit that code instead.
The CSS styling could do with significant improvement as well, but for now just get the PHP uploaded.

chrisn02 reviewed (changes requested):
Some change needed. The version of login.php seems to be outdated as it's the same as one of the versions that didn't work when I remember it working at one point. CSS also needs work, remember to also match the colours to the main theme of the rest of the site.

gchodancar19 committed:
Pull requests

gchodancar19 committed:
Login page works on local host

Jforjo reviewed (changes requested):
You should significantly improve your CSS.
You should also have both a login page and a signup page which a user go switch between.
Also, you should have another function to create a user, because right now you only have a function to log the user in.
Finally, see individual line comments for more feedback.

Jforjo commented on auth.php:
```diff
@@ -1,67 +1,27 @@
+if ($data === false) {
+    header('Location: error.html');
+} else {
+    header('Location: success.html');
```
This should link to the dashboard instead.
This should also created a session variable called "userID".

Jforjo commented on auth.php:
```diff
@@ -1,67 +1,27 @@
+    $conn = null;
+    if ($data === null || $data === false) return false;
+    if (!password_verify($password, $data['password'])) return false;
+    return array(
```
This is not needed as it already returns in this format.

Jforjo commented on signup.php:
```diff
@@ -1,5 +1,67 @@
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <link rel="stylesheet" href="stylesheet.css">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Document</title>
```
The title "Document" isn't suitable for this page.

Jforjo commented on auth.php:
```diff
@@ -1,67 +1,27 @@
+$data = validateUser($_POST['email'], $_POST['password']);
+
+if ($data === false) {
+    header('Location: error.html');
```
This should redirect you back to the login page with the `?e=` in the URL with it's corresponding error code.

Jforjo commented on signup.php:
```diff
@@ -1,5 +1,67 @@
+<body>
+
+<form method= "POST"action="auth.php">
+<h2>Login</h2>
```
This should say "Signup".

Jforjo commented on signup.php:
```diff
@@ -1,5 +1,67 @@
+</head>
+<body>
+
+<form method= "POST"action="auth.php">
```
Formatting. Should say `<form method="POST" action="auth.php">`.

Jforjo commented on signup.php:
```diff
@@ -1,5 +1,67 @@
+<form method= "POST"action="auth.php">
+<h2>Login</h2>
+<label for="FirstName">FirstName:</label>
+<input name="FirstName" id="FirstName" type="FirstName" placeholder="John" required>
```
Doesn't need a placeholder, especially an example name.

Jforjo commented on signup.php:
```diff
@@ -1,5 +1,67 @@
+<input name="FirstName" id="FirstName" type="FirstName" placeholder="John" required>
+
+<label for="LastName">LastName:</label>
+<input name="LastName" id="LastName" type="LastName" placeholder="Ward" required>
```
Doesn't need a placeholder, especially an example name.

Jforjo commented on signup.php:
```diff
@@ -1,5 +1,67 @@
+
+<label for="email">Email:</label>
+<input name="email" id="email" type="email" placeholder="eg test@test.com" required>
+<input name="email" id="email" type="email" placeholder="eg bobsmith@gmail.com" required>
```
There shouldn't be 2 email inputs.

Jforjo commented on signup.php:
```diff
@@ -1,5 +1,67 @@
+<input name="password" id="password" type="password" placeholder="password" required>
+
+<label for="Student Number">Student Number:</label>
+<input name="Student Number" id="Student Number" type="Student Number" placeholder="Student Number" required>
```
"Student Number" is not a valid input type.
You also shouldn't use spaces in IDs or names.

Jforjo commented on signup.php:
```diff
@@ -1,5 +1,67 @@
+<input name="Student Number" id="Student Number" type="Student Number" placeholder="Student Number" required>
+
+<label for="Course Title">Course Title:</label>
+<input name="Course Title" id="Course Title" type="Course Title" placeholder="Course Title" required>
```
"Course Title" is not a valid input type.
You also shouldn't use spaces in IDs or names.

Jforjo commented on signup.php:
```diff
@@ -1,5 +1,67 @@
+<input name="Course Title" id="Course Title" type="Course Title" placeholder="Course Title" required>
+
+
+<button type="submit"> You are sign up!</button>
```
This should say something like "Signup".

Jforjo commented on signup.php:
```diff
@@ -1,5 +1,67 @@
+if(isset($_GET["e"]))
+{
+    
+if ($_GET["e"]==5)
```
Why does this start at 5??

Jforjo commented on signup.php:
```diff
@@ -1,5 +1,67 @@
+    
+if ($_GET["e"]==5)
+{
+    echo "<span class= 'error'>Invalid Username or Password.</span>";
```
This errors should be outputted below the form title and above the first input field.

Jforjo commented on login.php:
```diff
@@ -0,0 +1,53 @@
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <link rel="stylesheet" href="stylesheet.css">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Document</title>
```
This should say something like "Login".

Jforjo commented on login.php:
```diff
@@ -0,0 +1,53 @@
+<label for="password">Password:</label>
+<input name="password" id="password" type="password" placeholder="password" required>
+
+<button type="submit"> You are logged in!</button>
```
This should just say "Log in".

Jforjo commented on signup.php:
```diff
@@ -0,0 +1,67 @@
+</head>
+<body>
+
+<form method= "POST"action="auth.php">
```
This is the signup form and therefore should not link to the form used for logging in.

Jforjo reviewed (changes requested):
Added more comments.
Include function to create a student for the signup page.

gchodancar19 committed:
All the changes are been done

chrisn02 reviewed (changes requested):
The login page has been improved, but still needs some styling changes. There's a lot of redundant data in the css file, such as styling for a gallery, so make sure to remove that when working on it.

gchodancar19 committed:
Started styling please have a look at that.

Jforjo commented on login.php:
```diff
@@ -48,6 +35,21 @@
 }
 
 ?>
+<div class= "formcontainer">
```
This form container is not needed.

Jforjo commented on login.php:
```diff
@@ -48,6 +35,21 @@
+<label for="password">Password:</label>
+<input name="password" id="password" type="password" placeholder="password" required>
+
+<button type="submit"> login!</button>
```
You should capitalise the word "Login" and remove the exclamation mark.

Jforjo commented on login.php:
```diff
@@ -48,6 +35,21 @@
+
 
 </body>
 </html>
```
These 2 closing tags should not be here.

Jforjo commented on signup.php:
```diff
@@ -56,5 +36,26 @@
+<input name="password" id="password" type="password" placeholder="password" required>
+
+
+<button type="signup">signup!</button>
```
You should capitalise the word "Signup" and remove the exclamation mark.

Jforjo commented on signup.php:
```diff
@@ -56,5 +36,26 @@
+
+
 </body>
 </html>
```
These closing tags should not be here.

Jforjo commented on stylesheet.css:
```diff
@@ -1,282 +1,61 @@
 body {
```
You should have something like
```
* {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
    font-family: Arial, Helvetica, sans-serif;
}
```
at the top of the CSS.

Jforjo commented on stylesheet.css:
```diff
@@ -1,282 +1,61 @@
+  width: 100%;
+  height: 100vh;
+  margin-top: 20px;
+  margin-bottom: 20px;
```
You shouldn't have a margin on the body.

Jforjo commented on stylesheet.css:
```diff
@@ -1,282 +1,61 @@
+  height: 100vh;
+  margin-top: 20px;
+  margin-bottom: 20px;
+  overflow-x: hidden;
```
This overflow isn't needed.

Jforjo commented on stylesheet.css:
```diff
@@ -1,282 +1,61 @@
+  overflow-x: hidden;
+  font-family: "Segoe UI", Tahoma, Geneva, Verdana, sans-serif;
+  color:white;
+}
```
To centre a single item on a page you can simply add `display: grid;` and `place-items: center;`.

Jforjo commented on stylesheet.css:
```diff
@@ -1,282 +1,61 @@
+  align-items: center;
+}
+
+body header {
```
Your pages do not have a header tag, so remove all CSS with `body header` in.

Jforjo commented on stylesheet.css:
```diff
@@ -1,282 +1,61 @@
+  text-decoration: none;
+}
+
+.formcontainer{
```
Change `.formcontainer` to `form`

Jforjo commented on stylesheet.css:
```diff
@@ -1,282 +1,61 @@
+}
+
+.formcontainer{
+  width: 80%;
```
You should set a specified width instead of a percentage, so your form isn't really wide. 256px would be a nice size.

Jforjo commented on stylesheet.css:
```diff
@@ -1,282 +1,61 @@
+  width: 80%;
+  display: flex;
+  justify-content: center;
+  align-items: center;
```
Remove both `justify-content: center;` and `align-items: center;`.

Jforjo commented on stylesheet.css:
```diff
@@ -1,282 +1,61 @@
+  justify-content: center;
+  align-items: center;
+  box-shadow: 0px 5px 15px 0px rgba(0, 0, 0, 0.43);
+  border-bottom: 2px solid gray;
```
Remove this line.

Jforjo commented on stylesheet.css:
```diff
@@ -1,282 +1,61 @@
+  align-items: center;
+  box-shadow: 0px 5px 15px 0px rgba(0, 0, 0, 0.43);
+  border-bottom: 2px solid gray;
+  flex-wrap: wrap;
```
`flex-wrap: wrap;` is useless if you have `flex-direction: column`. So remove the wrap.

Jforjo commented on stylesheet.css:
```diff
@@ -1,282 +1,61 @@
+  flex-direction: column;
+  border: 1px solid white;
+  
+
```
You should add some more CSS to add a `margin-bottom: 16px;` to all inputs, so they are separated out.

Jforjo reviewed (changes requested):
Read comments below.

gchodancar19 committed:
Login page completed

gchodancar19 committed:
started to create user for my signup page.

In order to get my signup page working I need to create a user where I need to know will the code be similar to my auth page.

chrisn02 reviewed (approved):
The login page now works since a user can successfully log in and be taken to the dashboard. Remember to add some more styling to the page however. There also seems to be code for a signup page, which I feel would be better suited in its own branch as that page is not finished yet.

Jforjo commented on stylesheet.css:
```diff
@@ -0,0 +1,36 @@
+  margin-bottom: 16px;
+
+}
+
```
Add some margin bottom to the inputs in the form.

gchodancar19 replied:
Can you please explain this in detail.

Jforjo replied:
Currently the form inputs/labels are too close together, so adding `margin-bottom` to your inputs would solve this.

Jforjo commented on stylesheet.css:
```diff
@@ -0,0 +1,36 @@
+  box-shadow: 0px 5px 15px 0px rgba(0, 0, 0, 0.43);
+  flex-direction: column;
+  border: 1px solid white;
+  margin-bottom: 16px;
```
Considering how there is nothing below the form, remove the `margin-bottom`.

Jforjo commented on stylesheet.css:
```diff
@@ -0,0 +1,36 @@
+  display: flex;
+  box-shadow: 0px 5px 15px 0px rgba(0, 0, 0, 0.43);
+  flex-direction: column;
+  border: 1px solid white;
```
Add a padding to separate the border from the content inside.

Jforjo commented on stylesheet.css:
```diff
@@ -0,0 +1,36 @@
+  place-items: center;
+}
+
+.login page {
```
There is no `<page>` tag, so you can just remove this section.

Jforjo commented on stylesheet.css:
```diff
@@ -0,0 +1,36 @@
+body {
+  background-color: #181818;
+  width: 100%;
+  height: 100vh;
```
A body should have `min-height`, not `height`.
This is because when you have too much content in the body then it will not expand.

Jforjo commented on signup.php:
```diff
@@ -0,0 +1,59 @@
+<input name="password" id="password" type="password" placeholder="password" required>
+
+
+<button type="signup">Signup</button>
```
The `button` type "signup" is invalid. Use `type="submit"`.

Jforjo commented on signup.php:
```diff
@@ -0,0 +1,59 @@
+<input name="FirstName" id="FirstName" type="FirstName" required>
+
+<label for="LastName">LastName:</label>
+<input name="LastName" id="LastName" type="LastName" required>
```
Remove the capital 'L' in all instances and replace it with a lowercase 'l'.

Jforjo commented on signup.php:
```diff
@@ -0,0 +1,59 @@
+?>
+
+<label for="FirstName">FirstName:</label>
+<input name="FirstName" id="FirstName" type="FirstName" required>
```
Remove the capital 'F' in all instances and replace it with a lowercase 'f'.

Jforjo commented on login.php:
```diff
@@ -0,0 +1,51 @@
+<label for="password">Password:</label>
+<input name="password" id="password" type="password" placeholder="password" required>
+
+<button type="submit"> Login</button>
```
Remove the space before "Login".

Jforjo commented on createuser.php:
```diff
@@ -0,0 +1,5 @@
+require_once('php/connection.php');
+function validateUser($email, $password) {
+    
+}
```
This should not be here.

Jforjo reviewed (changes requested):
You shouldn't keep requesting reviews, especially with unfinished files (`createuser.php`).
However, see comments for some changes you can make.

Jforjo commented on auth.php:
```diff
@@ -0,0 +1,26 @@
+    $conn = null;
+    if ($data === null || $data === false) return false;
+    if (!password_verify($password, $data['password'])) return false;
+   return$data;
```
Add a check to see if the user's 'state' is 'active', and if it isn't then don't allow them to log in.

gchodancar19 replied:
Can you please state exactly what I need to do.

gchodancar19 replied:
Because I struggling to get that done.

Jforjo replied:
```php
if ($data['state'] !== 'active') return false;
```

gchodancar19 committed:
Createuser.php page

I have created createuser.php page and it works can you make all suggestions.

Jforjo commented on createuser.php:
```diff
@@ -1,5 +1,51 @@
+`email` = '$email',
+`password` = '$password'
+WHERE
+`users`.`userid` = $uid";
```
This entire SQL statement is wrong.
Firstly, 'update' modifies an existing user. You want 'insert' to create a new user. __BUT DON'T DO THIS__
There is already a function in SQL called 'CreateUser' that will create a user with some given parameters. Your SQL should look something like `CALL CreateUser(:userID, :firstname, :lastname, :email, :password);`

Jforjo commented on createuser.php:
```diff
@@ -1,5 +1,51 @@
+{
+    header("Location: login.php?e=4");
+ die("You are not Signed in.");  }  
+}
```
What is the purpose of this IF statement?
Just remove it entirely.

Jforjo commented on createuser.php:
```diff
@@ -1,5 +1,51 @@
+WHERE
+`users`.`userid` = $uid";
+
+if(mysqli_query($db_connect,$sql))
```
This will not work.
Look to your `auth.php` file as an example of what your code should look like, but it will need changing.

Jforjo commented on createuser.php:
```diff
@@ -1,5 +1,51 @@
+    header("Location: login.php");
+}
+
+echo mysqli_error($db_connect);
```
Not needed.

Jforjo commented on createuser.php:
```diff
@@ -1,5 +1,51 @@
+
+else { die("No user selected!");}
+?>
+<h2>Edit user:       <?php echo $result["email"]; ?>              </h2>
```
What is the purpose of this?
Remove this.

Jforjo commented on createuser.php:
```diff
@@ -1,5 +1,51 @@
+`firstname` = '$f_name',
+`lastname` = '$l_name',
+`email` = '$email',
+`password` = '$password'
```
These variables don't exist.

Jforjo commented on createuser.php:
```diff
@@ -1,5 +1,51 @@
+    $uid = $_GET["userid"];
+    $sql = "SELECT * FROM `users` WHERE `userID` = '$uid' LIMIT 1"; //SQL Query
+    $query = mysqli_query($db_connect,$sql);//get the results
+    $result = mysqli_fetch_assoc($query);
```
This would part is not needed.

Jforjo commented on createuser.php:
```diff
@@ -1,5 +1,51 @@
+
+}
+
+else { die("No user selected!");}
```
Not needed.

Jforjo commented on createuser.php:
```diff
@@ -1,5 +1,51 @@
\ No newline at end of file
+
+session_start();
+if(isset($_SESSION["auth"])){  
```
Not needed.

Jforjo reviewed (changes requested):
See comments.

gchodancar19 committed:
Regarding my signup page fixed and working.

My signup page does not work on xampp but it might work on plesk.
